### PR TITLE
Add CXX_FLAGS_BY_COMPILER_FRONTEND function

### PR DESCRIPTION
**Pull request by jorisv**
This function allow to choose different compiler options or definitions based on the compiler frontend

### CODE REVIEW

**Comment by jorisv** [comment]
We can maybe use this macro in _SETUP_PROJECT_WARNINGS

File: compiler.cmake [hunk 1]
```diff
@@ -19,51 +19,140 @@ macro(_SETUP_PROJECT_WARNINGS)
...
+
+function(CXX_FLAGS_BY_COMPILER_FRONTEND)
+  set(options FILTER)
+  set(oneValueArgs OUTPUT)
+  set(multiValueArgs GNU MSVC)
+  cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}"
+                        ${ARGN})
+
+  # Before CMake 3.14, the CMAKE_CXX_COMPILER_FRONTEND_VARIANT doesn't exists.
+  # Before CMake 3.26, the CMAKE_CXX_COMPILER_FRONTEND_VARIANT doesn't exists
```
Comment (nim65s) [inline compiler.cmake:115]: There is something strange with those 2 comments

**Review by nim65s — APPROVED** [review]
Looks good to me, except for a misleading comment

File: compiler.cmake (hunk 1 above)
Comment (jorisv) [reply compiler.cmake:115]: The comment line 115 continue line 116. The cmake-format has split the line.

I can rewrite it to make it clearer.

**Pull request merged**